wgengine/magicsock/magicsock.go

```diff
@@ -952,6 +957,27 @@ func (c *Conn) determineEndpoints(ctx context.Context) ([]tailcfg.Endpoint, erro
 		addAddr(ap, tailcfg.EndpointExplicitConf)
 	}
 
+	// If we're on a cloud instance, we might have a public IPv4 or IPv6
+	// address that we can be reached at. Find those, if they exist, and
+	// add them.
+	if addrs, err := c.cloudInfo.GetPublicIPs(ctx); err == nil {
+		var port4, port6 uint16
+		if addr := c.pconn4.LocalAddr(); addr != nil {
+			port4 = uint16(addr.Port)
+		}
+		if addr := c.pconn6.LocalAddr(); addr != nil {
+			port6 = uint16(addr.Port)
+		}
+
+		for _, addr := range addrs {
+			if addr.Is4() && port4 > 0 {
+				addAddr(netip.AddrPortFrom(addr, port4), tailcfg.EndpointLocal)
```

Comment:
create a new type? `Local` seems wrong for `Public` but I don't particularly know how we decide these, should probably add comments to the enums to make it clearer

Reply:
I'll follow up with a PR to add docs. I think `EndpointLocal` is correct here, since that's the same type that would be used if the address was attached directly to the interface. We don't use these types for much anyway, though, so we could add a new type if we really wanted.

Reply:
We do use them in control for deciding which endpoints to proactively push down to peers. But considering that almost all endpoint exchange happens peer-to-peer over disco over DERP in CallMeMaybe anyway, it kinda doesn't matter. It's just interesting for stats, really.

So this is fine for now. But new type would be good too.